src/Aquifer.Common/Services/Caching/CachingVersificationService.cs

```diff
@@ -0,0 +1,42 @@
+using Aquifer.Data;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Aquifer.Common.Services.Caching;
+
+public interface ICachingVersificationService
+{
+    Task<Dictionary<(int bibleVerseId, char? bibleVersePart), (int baseVerseId, char? baseVersePart)>> GetVersificationsByBibleIdAsync(
```

Comment:
I suggest returning a `ReadOnlyDictionary`.  Callers should never modify this return value because if they do it will modify the cached dictionary and impact all future callers.  Using a `ReadOnlyDictionary` incurs a small performance cost but protects against that.

Reply:
good catch.